be/src/fs/fs_starlet.cpp

```diff
@@ -504,6 +504,29 @@ class StarletFileSystem : public FileSystem {
         return to_status((*fs_st)->drop_cache(pair.first));
     }
 
+    Status delete_files(const std::vector<std::string_view>& paths) override {
+        std::vector<std::string> parsed_paths;
+        parsed_paths.resize(paths.size());
+        std::shared_ptr<staros::starlet::fslib::FileSystem> fs = nullptr;
+        for (size_t i = 0; i < paths.size(); ++i) {
+            ASSIGN_OR_RETURN(auto pair, parse_starlet_uri(paths[i]));
+            if (i == 0) {
+                auto fs_st = get_shard_filesystem(pair.second);
+                if (!fs_st.ok()) {
+                    return to_status(fs_st.status());
+                }
+                fs = *fs_st;
```

Comment:
shouldn't report error if in case get different fs from `get_shard_filesystem` ?